### ISSUE TITLE
Crash in Niagara Blueprint

### ISSUE THREAD

**Issue by NicolasPanda** [fpwong/BlueprintAssistWiki#179]
This issue only occurs in the Blueprint Editor of Niagara.
When using Blueprint Assist, I am unable to change the names of variables. Additionally, when I create a node 'Set Num Cells' _(Or any node)_, the editor crashes.

### Steps to Reproduce:
1. Use Blueprint Assist in Unreal 5.4.
2. Attempt to change the name of any variable.
3. Create a 'Set Num Cells' node in the Niagara Blueprint Editor.

### Expected Behavior:
Variable names should be changeable, and creating a 'Set Num Cells' node should not crash the editor.

### Error Message:
```mathematica
Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:C:\Program Files\Epic Games\UE_5.4\Engine\Source\Runtime\Core\Public\Containers\Array.h] [Line: 758]
Array index out of bounds: 0 into an array of size 0
```
### Actual Behavior:
The editor crashes with the following error message when attempting to create a 'Set Num Cells' node:
```mathematica
Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:C:\Program Files\Epic Games\UE_5.4\Engine\Source\Runtime\Core\Public\Containers\Array.h] [Line: 758]
Array index out of bounds: 0 into an array of size 0
UnrealEditor_BlueprintAssist!FBAGraphHandler::LinkExecWhenCreatedFromParameter() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistGraphHandler.cpp:321]
UnrealEditor_BlueprintAssist!FBAGraphHandler::TryInsertNewNode() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistGraphHandler.cpp:265]
UnrealEditor_BlueprintAssist!FBAGraphHandler::OnNodesAdded() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistGraphHandler.cpp:947]
UnrealEditor_BlueprintAssist!FBAGraphHandler::DetectGraphChanges() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistGraphHandler.cpp:890]
UnrealEditor_BlueprintAssist!TBaseRawMethodDelegateInstance<0,FBAGraphHandler,void __cdecl(void),FDefaultDelegateUserPolicy>::ExecuteIfSafe() [C:\Program Files\Epic Games\UE_5.4\Engine\Source\Runtime\Core\Public\Delegates\DelegateInstancesImpl.h:534]
UnrealEditor_BlueprintAssist!FBADelayedDelegate::Tick() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistDelayedDelegate.cpp:19]
UnrealEditor_BlueprintAssist!FBAGraphHandler::Tick() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistGraphHandler.cpp:456]
UnrealEditor_BlueprintAssist!FBATabHandler::Tick() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistTabHandler.cpp:53]
UnrealEditor_BlueprintAssist!FBAInputProcessor::Tick() [C:\Users\UserName\ProjectName\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistInputProcessor.cpp:100]
UnrealEditor_Slate
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll
```
### Additional Information:
LoginId: (replaced for privacy)
EpicAccountId: (replaced for privacy)

### Additional Context:
This issue has consistently occurred only in the Niagara Blueprint Editor while using Blueprint Assist. Any guidance or a fix for this issue would be greatly appreciated.

**Thank you.**

**Comment by fpwong** [comment]
Hi, thanks for letting me know. I was able to replicate this and should have a fix in the next version.

As a workaround for now, you can go to Editor Preferences and disable the setting `AlwaysConnectExecutionFromParameter`, which should avoid calling the function that triggers the crash.

**Comment by fpwong** [comment]
Fixed in 4.2.16
